libsql-server/src/namespace/mod.rs

```diff
@@ -764,6 +765,21 @@ impl<M: MakeNamespace> NamespaceStore<M> {
     ) -> crate::Result<MetaStoreHandle> {
         self.with(namespace, |ns| ns.db_config_store.clone()).await
     }
+
+    pub async fn jwt_key(
```

Comment:
So I actually don't really like this approach, to me I envisioned that this would exist on the `Namespace` struct itself and you would get access to it via `with`. So once you have parsed the namespace out of the http header, you check the store for it, it then gives you a `Namespace<T>`  and on that namespace you can then access the config. There is no need to have a separate function that threads through namespaces since its all type checked via the store/namespace struct. This imo provides a really nice encapsulation where you can only access the config from the `Namespace` struct that is tied to that namespace.

We can keep this for now but its quite leaky imo and we should fix this in the future and track that.

Reply:
Moved jwt_key to Namespace.